src/System.Threading.Channels/tests/Stress.cs

```diff
@@ -0,0 +1,423 @@
+            yield return new object[] { Channel.CreateBounded<int>(BoundedDropWriteSingleWriterOptions), BoundedDropWriteSingleWriterOptions };
+            yield return new object[] { Channel.CreateBounded<int>(BoundedDropWriteSingleReaderWriterOptions), BoundedDropWriteSingleReaderWriterOptions };
+
+            //
+            // Bounded Channel ( AllowSynchronousContinuations,  BoundedChannelFullMode.DropWrite)
+            //
+            yield return new object[] { Channel.CreateBounded<int>(BoundedSynchronousDropWriteOptions), BoundedSynchronousDropWriteOptions };
+            yield return new object[] { Channel.CreateBounded<int>(BoundedSynchronousDropWriteSingleReaderOptions), BoundedSynchronousDropWriteSingleReaderOptions };
+            yield return new object[] { Channel.CreateBounded<int>(BoundedSynchronousDropWriteSingleWriterOptions), BoundedSynchronousDropWriteSingleWriterOptions };
+            yield return new object[] { Channel.CreateBounded<int>(BoundedSynchronousDropWriteSingleReaderWriterOptions), BoundedSynchronousDropWriteSingleReaderWriterOptions };
+
+            //
+            // Unbuffered Channel
+            //
+            yield return new object[] { Channel.CreateUnbuffered<int>(UnbufferedDefaultOptions), UnbufferedDefaultOptions };
+            yield return new object[] { Channel.CreateUnbuffered<int>(UnbufferedSingleReaderOptions), UnbufferedSingleReaderOptions };
+            yield return new object[] { Channel.CreateUnbuffered<int>(UnbufferedSingleReaderWriterOptions), UnbufferedSingleReaderWriterOptions };
+            yield return new object[] { Channel.CreateUnbuffered<int>(UnbufferedSingleWriterOptions), UnbufferedSingleWriterOptions };
+
+            //
+            // Unbuffered Channel (AllowSynchronousContinuations)
+            //
+            yield return new object[] { Channel.CreateUnbuffered<int>(UnbufferedSynchronousOptions), UnbufferedSynchronousOptions };
+            yield return new object[] { Channel.CreateUnbuffered<int>(UnbufferedSynchronousSingleReaderOptions), UnbufferedSynchronousSingleReaderOptions };
+            yield return new object[] { Channel.CreateUnbuffered<int>(UnbufferedSynchronousSingleReaderWriterOptions), UnbufferedSynchronousSingleReaderWriterOptions };
+            yield return new object[] { Channel.CreateUnbuffered<int>(UnbufferedSynchronousSingleWriterOptions), UnbufferedSynchronousSingleWriterOptions };
+            yield return new object[] { Channel.CreateUnbuffered<int>(UnbufferedSynchronousOptions), UnbufferedSynchronousOptions };
+        }
+
+        const int MAX_NUMBER_TO_WRITE_TO_CHANNEL = 1_000_000;
+        private static readonly int MaxTaskCounts = Math.Max(2, Environment.ProcessorCount);
+
+        [OuterLoop]
+        [Theory]
+        [MemberData(nameof(TestData))]
+        public void RunInStressMode(Channel<int> channel, ChannelOptions options)
```

Comment:
What are you trying to stress here specifically?

Reply:
I am trying to stress simultaneous read and write operation on the channels with different channels options.